File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.java
class Solution {
    private PriorityQueue<Integer> minHeap;
    private int k;

    public int findKthLargest(int[] nums, int k) {
        this.k = k;
        minHeap = new PriorityQueue<>();
        for (int num : nums) {
            add(num);
        }

        return minHeap.peek();
    }

    public int add(int val) {
        if (minHeap.size() < k) {
            minHeap.offer(val);
        } else if (val > minHeap.peek()) {
            minHeap.poll();
            minHeap.offer(val);
        }
        return minHeap.peek();
    }

}

File: Data Structures & Algorithms/concatenation-of-array/submission-0.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [None] * len(nums) * 2
        for i, num in enumerate(nums):
            ans[i] = nums[i]
            ans[i+n] = nums[i]
        return ans